docs/api.rst

```diff
@@ -4348,6 +4348,252 @@ Getting AdEx historical data
+   :statuscode 502: An external service used in the query such as the graph node could not be reached or returned unexpected response.
+
+Getting Balancer trades
+=========================
+
+.. http:get:: /api/(version)/blockchains/ETH/modules/balancer/history/trades
+
+   Doing a GET on the Balancer trades history resource will return the history of all Balancer trades.
+
+   .. note::
+      This endpoint is only available for premium users
+
+   .. note::
+      This endpoint can also be queried asynchronously by using ``"async_query": true``
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      GET /api/1/blockchains/ETH/modules/balancer/history/trades HTTP/1.1
+      Host: localhost:5042
+
+   :reqjson bool async_query: Boolean denoting whether this is an asynchronous query or not
+
+   **Example Response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/json
+
+      {
+        "result": {
+            "0x029f388aC4D5C8BfF490550ce0853221030E822b": [
+                {
+                    "address": "0x029f388aC4D5C8BfF490550ce0853221030E822b",
+                    "amount": "0.075627332013165531",
+                    "base_asset": "WETH",
+                    "fee": "0",
+                    "fee_currency": "QNT",
+                    "location": "balancer",
+                    "pair": "WETH_QNT",
+                    "quote_asset": "QNT",
+                    "rate": "0.02194014031410883771422129499",
+                    "swaps": [
+                        {
+                            "amount0_in": "3.446984883890308608",
+                            "amount0_out": "0",
+                            "amount1_in": "0",
+                            "amount1_out": "0.075627332013165531",
+                            "from_address": "0x0000000000007F150Bd6f54c40A34d7C3d5e9f56",
+                            "location": "balancer",
+                            "log_index": 37,
+                            "to_address": "0x6545773483142Fd781023EC74ee008d93aD5466B",
+                            "token0": "QNT",
+                            "token1": "WETH",
+                            "tx_hash": "0x9a5c2c73762ef2e8af326e7b286488a4b238b9855d3fd749370bb3074aabf6e5"
+                        }
+                    ],
+                    "timestamp": 1606924530,
+                    "trade_id": "0x9a5c2c73762ef2e8af326e7b286488a4b238b9855d3fd749370bb3074aabf6e5-0",
+                    "trade_type": "buy",
+                    "tx_hash": "0x9a5c2c73762ef2e8af326e7b286488a4b238b9855d3fd749370bb3074aabf6e5"
+                },
+                {
+                    "address": "0x029f388aC4D5C8BfF490550ce0853221030E822b",
+                    "amount": "3.214606868598057153",
+                    "base_asset": "MKR",
+                    "fee": "0",
+                    "fee_currency": "WETH",
+                    "location": "balancer",
```

Comment:
Did you mean the `location` in the AMMTrade or the AMMSwap? For the former please, let me know if it is safe to be removed (as it has to behave like a trade, therefore I would understand it requires a location). About the latter it would make sense to do not have the location serialized (despite that Uniswap serializes swaps with location as well). I can remove it.

Reply:
This is from `class Trade`:
```python
    def serialize(self) -> Dict[str, Any]:
        """Serialize the trade into a dict"""
        return {
            'timestamp': self.timestamp,
            'location': str(self.location),
            'pair': self.pair,
            'trade_type': str(self.trade_type),
            'amount': str(self.amount),
            'rate': str(self.rate),
            'fee': str(self.fee),
            'fee_currency': self.fee_currency.identifier,
            'link': self.link,
            'notes': self.notes,
        }
```

Reply:
I know. But the trade is returned from the all trades endpoint. Here you are returning it from a different endpoint unique to balancer so it can be omitted. Anyway this is minor ... just forget it.